### PR TITLE
Expose package library when running tests

### LINKED ISSUE #4896: Stack 2.1.1 doesn't set package environments like it used to
### General summary/comments (optional)

With previous versions of Stack, `stack test` set the package environment so the current package was loaded. This is useful for [IHaskell](https://github.com/gibiansky/IHaskell) which exercises the GHC API.

Stack 2.1.1 doesn't do this, and so I'm getting a lot of errors and a failed test suite.

### Steps to reproduce

1. `git clone https://github.com/gibiansky/IHaskell`
1. `cd IHaskell`
1. `stack test`

### Expected

`ihaskell-0.10.0.0: Test suite hspec passed`

This is with version `1.9.3 x86_64 hpack-0.31.2`.

### Actual

https://travis-ci.org/gibiansky/IHaskell/jobs/548734133

### Stack version

```
$ stack --version
Version 2.1.1, Git revision f612ea85316bbc327a64e4ad8d9f0b150dc12d4b (7648 commits) x86_64 hpack-0.31.2
```

### Method of installation

* Official binary, downloaded from stackage.org or fpcomplete's package repository

### PR DESCRIPTION
**Pull request by qrilka**
Note: Documentation fixes for https://docs.haskellstack.org/en/stable/ should target the "stable" branch, not master.

Please include the following checklist in your PR:

* [ ] Any changes that could be relevant to users have been recorded in the ChangeLog.md
* [ ] The documentation has been updated, if necessary.

Fixes #4896 
@snoyberg this appears to be just a quickfix. I'm not yet sure that it's the correct way.
I didn't quite understand the motivation of excluding package library from dependencies in the commit 9ceb409a887bfcb1c0a09c3c783e409f9ecaed97  - does it make sense to try reverting that or it's just a workaround until we have proper component-based builds?

### CODE REVIEW

File: src/Stack/Build/Execute.hs [hunk 1]
```diff
@@ -1949,7 +1961,7 @@ singleTest topts testsToRun ac ee task installedMap = do
                         "package-db " <> fromString snapDBPath <> "\n" <>
                         "package-db " <> fromString localDBPath <> "\n" <>
                         foldMap (\ghcId -> "package-id " <> RIO.display (unGhcPkgId ghcId) <> "\n")
-                            (thGhcId:M.elems allDepsMap)
```
Comment (snoyberg) [inline src/Stack/Build/Execute.hs]: Why is `thGhcId` now excluded?

Comment (qrilka) [reply src/Stack/Build/Execute.hs]: what do you mean by "excluded"? I see it in `pkgGhcIdList ++ thGhcId:M.elems allDepsMap`

**Comment by qrilka** [comment]
Oh, sublibraries, yes, that was also one of my guesses
So then I just need to add a changelog and check integration tests as an extra safety measure, right?

File: src/Stack/Build/Execute.hs (hunk 1 above)
Comment (snoyberg) [reply src/Stack/Build/Execute.hs]: Sorry, I'm stupid, ignore me. Totally misread the code.

**Comment by snoyberg** [comment]
Correct. Let me know if you want me to kick off integration test runs on Azure.

**Comment by qrilka** [comment]
Integration tests failed because of `tar` issue and a problem with `stack clean` on Windows (@psibi should create an issue about it).
Otherwise this PR should be OK to merge

**Comment by gelisam** [comment]
Has this fix been later reverted? I get the correct behaviour when I checkout and build commit 61edf6b, but not when I install the latest version of stack via `stack upgrade --git`.

**Comment by snoyberg** [comment]
I think the fix is only on stable, and hasn't yet been merged to master.

**Comment by gelisam** [comment]
`stack upgrade --git --git-branch stable` did the trick, thanks!

**Pull request merged**